### ISSUE TITLE
False positive "incompatible type" on functions inside classes 

### ISSUE THREAD

**Issue by RandomGuyNumber731** [python/mypy#17545]
**Bug Report**

calling functions with variables inside classes causes false positive "incompatible type"

**To Reproduce**

```python
class test:
    def new(n):
        print (n)


test.new(1)
```

**Expected Behavior**

im not sure if its intentional and im using python wrong or if the mypy fails at some point

**Actual Behavior**

it says `Argument 1 to "new" has incompatible type "int"; expected "test"`
although it seems to run normally for me

**Your Environment**

pretty much just plane vscode with python support and mypy

- Mypy version used: 1.10.1
- Python version used: 3.10.12

**Comment by iurisilvio** [comment]
I think your first arg should be `self`, not an int.

You have to make your method `@staticmethod` to work this way.

**Comment by hauntsaninja** [comment]
Yeah, the type annotation for self is inferred
